polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -363,13 +880,12 @@ struct TypeAlignOpLowering : public ConvertOpToLLVMPattern<TypeAlignOp> {
 
 void populatePolygeistToLLVMConversionPatterns(LLVMTypeConverter &converter,
                                                RewritePatternSet &patterns) {
-  // clang-format off
-  patterns.add<TypeSizeOpLowering>(converter);
-  patterns.add<TypeAlignOpLowering>(converter);
-  patterns.add<SubIndexOpLowering>(converter);
-  patterns.add<Memref2PointerOpLowering>(converter);
-  patterns.add<Pointer2MemrefOpLowering>(converter);
-  // clang-format on
+  patterns.add<TypeSizeOpLowering, TypeAlignOpLowering, SubIndexOpLowering,
+               Memref2PointerOpLowering, Pointer2MemrefOpLowering>(converter);
+  if (converter.getOptions().useBarePtrCallConv)
+    patterns.add<SubIndexBarePtrOpLowering, BareMemref2PointerOpLowering,
+                 BarePointer2MemrefOpLowering>(converter,
+                                               /*benefit*/ 2);
```

Comment:
Same for all cases: regular patterns are still added, but bare ptr patterns are added with a higher benefit (priority).